test/test.ts

```diff
@@ -1,23 +1,74 @@
-import axios from "axios";
-import server from "../src/index"; // Import the server setup
 import { describe, before, it, after } from "mocha"; // Mocha imports
-import assert from "assert";
+import { expect, assert } from "chai";
+import { setup } from "../src/setup";
+import { prisma } from "../src/setup-db";
+import { server, url } from "../src/setup-server";
+import axios from "axios";
 
 describe("GraphQL Server Test", function () {
-  let serverUrl: string;
-
   before(async function () {
-    serverUrl = "http://localhost:4000";
+    console.log("Starting setup");
+    await setup();
+    console.log("Setup complete");
   });
 
-  it("should return 'Hello world!' from the hello query", async function () {
-    const response = await axios.post(serverUrl, {
+  it("Returned 'Hello world!' from the hello query", async function () {
+    const response = await axios.post(url, {
       query: "{ hello }"
     });
 
     assert.equal(response.data.data.hello, "Hello world!");
   });
 
+  it("Created a new user with a mutation", async function () {
+    const response = await axios.post(url, {
+      query: `
+          mutation {
+            createUser(data: {
+              name: "User Test",
+              email: "test@example.com",
+              password: "123abc",
+              birthDate: "01-01-2000"
+            }) {
+              id
+              name
+              email
+              birthDate
+            }
+          }
+        `
+    });
+
+    expect(response.data).to.have.property("data");
+    expect(response.data.data).to.have.property("createUser");
+    expect(response.data.data.createUser).to.not.be.null;
+    expect(response.data.data.createUser.id).to.not.be.null;
+    assert.equal(response.data.data.createUser.name, "User Test");
+    assert.equal(response.data.data.createUser.email, "test@example.com");
+    assert.equal(response.data.data.createUser.birthDate, "01-01-2000");
```

Comment:
Para conferir a resposta de uma query, a abordagem mais completa seria utilzando o `to.be.deep.eq`, em que ele checa cada chave e valor dos objetos passados. Isso é bom porque se se tiver campos a mais, campos faltando ou campos com valores diferentes, o teste falha.